src/haxeLanguageServer/features/ExtractConstantFeature.hx

```diff
@@ -0,0 +1,130 @@
+package haxeLanguageServer.features;
+
+import haxe.io.Path;
+import sys.FileSystem;
+import haxeLanguageServer.helper.WorkspaceEditHelper;
+import tokentree.TokenTree;
+import tokentree.utils.TokenTreeCheckUtils;
+
+using tokentree.TokenTreeAccessHelper;
+
+class ExtractConstantFeature {
+	final context:Context;
+
+	public function new(context:Context) {
+		this.context = context;
+		#if debug
+		context.registerCodeActionContributor(extractConstant);
+		#end
+	}
+
+	function extractConstant(params:CodeActionParams):Array<CodeAction> {
+		var doc = context.documents.get(params.textDocument.uri);
+		try {
+			var fsPath:FsPath = params.textDocument.uri.toFsPath();
```

Comment:
Why do we need to check the file path or make assumptions about it being a file on disk at all? One nice thing about the tokentree-based features is that they generally also work on untitled files that aren't saved yet.

Reply:
I think I copied the check for `*.hx` files from someplace and never questioned it, might need some revision